WcaOnRails/app/models/wcif.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+class Wcif < ApplicationRecord
+  belongs_to :competition
+
+  # See https://github.com/thewca/worldcubeassociation.org/wiki/wcif
+  def json
+    json_db = JSON.parse(super || "{}")
+    json_db.merge(
+      "formatVersion" => "1.0",
+      "id" => competition.id,
+    )
+  end
+
+  def json=(new_json)
+    json = self.json.merge(new_json)
+    super(JSON.generate(json))
+  end
+
+  def json_hash
```

Comment:
Urg, I just realized that with the current hash implementation here, someone registering for a competition would invalidate your hash, which would prevent you from setting the WCIF for a competition.

I do not want to go down the rabbit hole of trying to solve this problem, so forget it, I'm removing this hash stuff from this PR. If people clobbering their own competition data becomes a problem is in the future, we'll discuss solutions then.